npm_and_yarn/lib/dependabot/npm_and_yarn/file_updater/pnpm_lockfile_updater.rb

```diff
@@ -201,6 +212,21 @@ def write_final_package_json_files
           end
         end
 
+        def raise_unsupported_platform_error(error_message, _pnpm_lock)
+          unless error_message.match(PLATFORM_PACAKGE_DEP) &&
+                 error_message.match(PLATFORM_VERSION_REQUIREMENT)
+            return
+          end
+
+          supported_version = error_message.match(PLATFORM_VERSION_REQUIREMENT)
+                                           .named_captures["supported_ver"].then { sanitize_message(_1) }
+          detected_version = error_message.match(PLATFORM_VERSION_REQUIREMENT)
+                                          .named_captures["detected_ver"].then { sanitize_message(_1) }
+
```

Comment:
Possible for improving readability this can changed as 
```
supported_version = error_message
                      .match(PLATFORM_VERSION_REQUIREMENT)
                      .named_captures["supported_ver"]
                      .then { |msg| sanitize_message(msg) }

detected_version = error_message
                     .match(PLATFORM_VERSION_REQUIREMENT)
                     .named_captures["detected_ver"]
                     .then { |msg| sanitize_message(msg) }
```